sdks/go/pkg/beam/runners/prism/internal/stage.go

```diff
@@ -194,7 +194,8 @@ progress:
 		case resp = <-b.Resp:
 			bundleFinished = true
 			if b.BundleErr != nil {
-				return b.BundleErr
+				// return b.BundleErr
+				panic("this should get triggered")
```

Comment:
Interestingly, somehow this does not seem to always raise an error? It only did in 3/4 runs - https://github.com/apache/beam/actions/runs/15685833595/job/44196132429

Reply:
The error there is very odd, though. It seems like an exception still does get raised, it just has no text? I'll rerun a few times to see if I can figure it out. It seems very surprising that this panic wouldn't work...

Reply:
It took 16 jobs, but eventually the panic in bundle.go didn't fire either. This is interesting because it fails much more consistently when we panic here

https://github.com/apache/beam/blob/679477947395746be01f9d1639283afd22d3d3a1/sdks/go/pkg/beam/runners/prism/internal/stage.go#L196-L198

than when we panic here

https://github.com/apache/beam/blob/e5e07c5100c0fdfdc6f864ee7b9fd69b04f6f51c/sdks/go/pkg/beam/runners/prism/internal/worker/bundle.go#L105-109

My current theory is that somehow in the stage execution loop, we're breaking before receiving the actual instruction response

Reply:
My take here is that:

https://github.com/apache/beam/blob/2e956263496564a614d216b7f985fa5a793baad7/sdks/go/pkg/beam/runners/prism/internal/stage.go#L209-L213

should just continue execution. This seems harmless since a failed progress report is not inherently problematic.

I think

https://github.com/apache/beam/blob/2e956263496564a614d216b7f985fa5a793baad7/sdks/go/pkg/beam/runners/prism/internal/stage.go#L225-L229

should return the error and fail the job. A failed split means that something important went wrong, and it is hard to know the current state of the bundle. I'd appreciate any input here though.

Reply:
I created a PR for this here - https://github.com/apache/beam/pull/35337